File: 0293-flip-game/0293-flip-game.java
class Solution {
    public List<String> generatePossibleNextMoves(String currentState) {
        List<String> nextPossibleStates = new ArrayList<>();
        for (int index = 0; index < currentState.length() - 1; ++index) {
            if (currentState.charAt(index) == '+' && currentState.charAt(index + 1) == '+') {
                String nextState = (currentState.substring(0, index) + "--" + currentState.substring(index + 2));
                nextPossibleStates.add(nextState);
            }
        }
        return nextPossibleStates;
    }
}

File: 0293-flip-game/0293-flip-game.cpp
class Solution {
public:
    vector<string> generatePossibleNextMoves(string currentState) {
        vector<string> nextPossibleStates;
        for (int index = 0; index < currentState.size() - 1; ++index) {
            if (currentState[index] == '+' and currentState[index + 1] == '+') {
                string nextState = (currentState.substr(0, index) + "--" + currentState.substr(index + 2));
                nextPossibleStates.push_back(nextState);
            }
        }
        return nextPossibleStates;
    }
};